src/shared/infra/repositories/reservation_repository_dynamo.py

```diff
@@ -0,0 +1,82 @@
+from src.shared.domain.entities.court import Court
+from src.shared.domain.repositories.reservation_repository_interface import IReservationRepository
+from src.shared.environments import Environments
+from src.shared.infra.dto.court_dynamo_dto import CourtDynamoDTO
+from typing import List, Optional
+from src.shared.infra.external.dynamo.datasources.dynamo_datasource import DynamoDatasource
+from src.shared.domain.enums.status_enum import STATUS
+from boto3.dynamodb.conditions import Key
+
+class ReservationRepositoryDynamo(IReservationRepository):
+    @staticmethod
+    def court_partition_key_format(number: int) -> str:
+        return f'court'
+    
+    @staticmethod
+    def court_sort_key_format(number: int) -> str:
+        return f'court#{number}'
+    
+
+    def __init__(self):
+        self.dynamo = DynamoDatasource(
+            endpoint_url=Environments.get_envs().endpoint_url,
+            dynamo_table_name=Environments.get_envs().dynamo_table_name,
+            region=Environments.get_envs().region,
+            partition_key=Environments.get_envs().dynamo_partition_key,
+            sort_key=Environments.get_envs().dynamo_sort_key,
+        )
+
+    def create_court(self, court: Court) -> Court:
+        item = CourtDynamoDTO.from_entity(court).to_dynamo()
+        resp = self.dynamo.put_item(item, partition_key=self.court_partition_key_format(court.number), sort_key=self.court_sort_key_format(court.number))
+
+        return court
+    
+    def get_court(self, number: int):
+        court = self.dynamo.get_item(partition_key=self.court_partition_key_format(number), sort_key=self.court_sort_key_format(number))
+
+        if "Item" not in court:
+            return None
+        
+        court_dto = CourtDynamoDTO.from_dynamo(court['Item'])
+        return court_dto.to_entity()
+        
+
+    def get_all_courts(self) -> list[Court]:
+
+        all_courts = []
+        for item in self.dynamo.get_all_items().get('Items'):
```

Comment:
faz a chamda do dynamo fora do 'for', da uma olhada na get_all_projects do PI

Reply:
feito